### ISSUE TITLE
ChangeZone message is not dispatched on initial zone change

### ISSUE THREAD

**Issue by quisquous** [ngld/OverlayPlugin#191]
Akurosia is trying to debug an issue where the initial zone change event is not being dispatched.

Here's what I know:
* cactbot uses OverlayPlugin's `ChangeZone` event
* this works just fine for me
* for Akurosia the first event is seemingly skipped and they have to change zones at least once to inform overlays about what zone they're in
* Akurosia can see this in the cactbot test overlay, where no `ChangeZone` event is delivered there
* I sent a logging build, and there is no DispatchAndCacheEvent in these cases
* There is an `01|` network log line in Akurosia's logs when this happens

This says to me that the FFXIV plugin is correctly sending the initial zone as a log line, but OverlayPlugin somehow isn't seeing it.  I've sent one additional logging build to Akurosia to print the first 40 lines that OverlayPlugin sees, and hopefully they can respond here with what that prints.

**Comment by Akurosia** [comment]
[Network_20704_20201227.log](https://github.com/ngld/OverlayPlugin/files/5745386/Network_20704_20201227.log)

Here is a new log created with the debug version quis sent me.
I am happy to debug any other suggestion :)

**Comment by quisquous** [comment]
Can you paste the overlayplugin log running that latest build?

**Comment by Akurosia** [comment]
```
[28.12.2020 11:34:38] Info: AssemblyResolver: Loaded: Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed
[28.12.2020 11:34:38] Info: AssemblyResolver: Loaded: websocket-sharp, Version=1.0.2.32519, Culture=neutral, PublicKeyToken=5660b08a1845a91e
[28.12.2020 11:34:38] Info: AssemblyResolver: Loaded: HtmlRenderer, Version=0.16.2.0, Culture=neutral, PublicKeyToken=null
[28.12.2020 11:34:38] Info: AssemblyResolver: Loaded: CefSharp, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[28.12.2020 11:34:38] Info: InitPlugin: PluginDirectory = E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld
[28.12.2020 11:34:38] Info: AssemblyResolver: Loaded: CefSharp.OffScreen, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[28.12.2020 11:34:38] Info: AssemblyResolver: Loaded: CefSharp.Core, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[28.12.2020 11:34:38] Info: AssemblyResolver: Loaded: OverlayPlugin.Core.resources, Version=0.16.2.0, Culture=de-DE, PublicKeyToken=null
[28.12.2020 11:34:38] Info: InitPlugin: Initialised.
[28.12.2020 11:34:44] Info: DEBUG: RegisterCachedEventType: ChangeZone
[28.12.2020 11:34:45] Info: cactbot: 0.21.10.0 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\CactbotOverlay.dll (dir: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot)
[28.12.2020 11:34:45] Info: OverlayPlugin: 0.16.2.0 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\OverlayPlugin.dll
[28.12.2020 11:34:45] Info: FFXIV Plugin: 2.0.7.4 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\FFXIV_ACT_Plugin.dll
[28.12.2020 11:34:45] Info: ACT: 3.5.0.273 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Advanced Combat Tracker.exe
[28.12.2020 11:34:45] Info: Parsing Plugin Language: de
[28.12.2020 11:34:45] Info: System Locale: de-DE
[28.12.2020 11:34:45] Info: cactbot user directory: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\user\
[28.12.2020 11:34:45] Info: Version: intl
[28.12.2020 11:34:45] Info: LoadAddons: CactbotOverlay.dll: Initialized Cactbot.PluginLoader
[28.12.2020 11:34:46] Info: WS: Warn: The current logging level has been changed to Info. 
[28.12.2020 11:34:47] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 11:34:47] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 11:34:47] Info: DEBUG: StateCallback: ChangeZone, eventCache: null
[28.12.2020 11:34:47] Info: DEBUG: StateCallback: ChangeZone, eventCache: null
[28.12.2020 11:34:47] Info: timeline: BrowserConsole: local user file: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\user\\raidboss.js (Source: file:///E:/FINAL%20FANTASY%20XIV%20-%20A%20Realm%20Reborn/AdditionalExtern/ACT/Plugins/OverlayPlugin_ngld/cactbot/dist/raidboss.bundle.js, Line: 1)
[28.12.2020 11:34:47] Info: timeline: BrowserConsole: local user file: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\user\\raidboss.css (Source: file:///E:/FINAL%20FANTASY%20XIV%20-%20A%20Realm%20Reborn/AdditionalExtern/ACT/Plugins/OverlayPlugin_ngld/cactbot/dist/raidboss.bundle.js, Line: 1)
[28.12.2020 11:34:47] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 11:34:47] Info: DEBUG: StateCallback: ChangeZone, eventCache: null
[28.12.2020 11:34:47] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 11:34:47] Info: DEBUG: StateCallback: ChangeZone, eventCache: null
[28.12.2020 11:34:47] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 11:34:47] Info: DEBUG: StateCallback: ChangeZone, eventCache: null
```

**Comment by Akurosia** [comment]
And after i teleport, i get following lines
```
[28.12.2020 11:44:38] Info: DEBUG: DispatchAndCacheEvent: ChangeZone: Mist
[28.12.2020 11:44:38] Info: DEBUG: DispatchEvent: ChangeZone, Mist
[28.12.2020 11:44:38] Info: timeline: BrowserConsole: Loading 00-misc/general.js (Source: file:///E:/FINAL%20FANTASY%20XIV%20-%20A%20Realm%20Reborn/AdditionalExtern/ACT/Plugins/OverlayPlugin_ngld/cactbot/dist/raidboss.bundle.js, Line: 1)
```

**Comment by quisquous** [comment]
Is that with the latest build I sent you? I sent a third debug build that also printed out the first 40 log lines.

**Comment by Akurosia** [comment]
i thought so, but i just made sure again to remove all of overlay plugin and unzip the version3 you send me (see followed):
```
[28.12.2020 19:57:22] Info: AssemblyResolver: Loaded: Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed
[28.12.2020 19:57:22] Info: AssemblyResolver: Loaded: websocket-sharp, Version=1.0.2.32519, Culture=neutral, PublicKeyToken=5660b08a1845a91e
[28.12.2020 19:57:22] Info: AssemblyResolver: Loaded: HtmlRenderer, Version=0.16.2.0, Culture=neutral, PublicKeyToken=null
[28.12.2020 19:57:22] Info: AssemblyResolver: Loaded: CefSharp, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[28.12.2020 19:57:22] Info: InitPlugin: PluginDirectory = E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld
[28.12.2020 19:57:22] Info: AssemblyResolver: Loaded: CefSharp.OffScreen, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[28.12.2020 19:57:22] Info: AssemblyResolver: Loaded: CefSharp.Core, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[28.12.2020 19:57:22] Info: AssemblyResolver: Loaded: OverlayPlugin.Core.resources, Version=0.16.2.0, Culture=de-DE, PublicKeyToken=null
[28.12.2020 19:57:22] Info: InitPlugin: Initialised.
[28.12.2020 19:57:27] Info: DEBUG: RegisterCachedEventType: ChangeZone
[28.12.2020 19:57:27] Info: DEBUG: LogLineHandler: line 1: 253|2020-12-28T19:57:21.8697496+01:00|FFXIV PLUGIN VERSION: 2.0.7.4|6aa0b78aa623df76730739380ec741f3
[28.12.2020 19:57:27] Info: cactbot: 0.21.10.0 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\CactbotOverlay.dll (dir: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot)
[28.12.2020 19:57:29] Info: OverlayPlugin: 0.16.2.0 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\OverlayPlugin.dll
[28.12.2020 19:57:29] Info: FFXIV Plugin: 2.0.7.4 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\FFXIV_ACT_Plugin.dll
[28.12.2020 19:57:29] Info: ACT: 3.5.0.273 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Advanced Combat Tracker.exe
[28.12.2020 19:57:29] Info: Parsing Plugin Language: de
[28.12.2020 19:57:29] Info: System Locale: de-DE
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 2: 249|2020-12-28T19:57:22.0446330+01:00|Language ID: 3, Disable Damage Shield: False, Disable Combine Pets: False, Parse Filter: Party, DoTCrits: False, RealDoTs: False, Parse Potency: False|9a80b322140bb72fe0a9b743ccbc3b98
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 3: 249|2020-12-28T19:57:22.0472349+01:00|Selected Process ID: 0, Disable Network Data: False, Enable HP Scan: True, Dump All Network Data: False, Disable Combat Log: False, Selected IP: , WinPcap: False, Socket Filter: True|bece02ac5c641d3832775f98c64da87c
[28.12.2020 19:57:29] Info: cactbot user directory: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\user\
[28.12.2020 19:57:29] Info: Version: intl
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 4: 251|2020-12-28T19:57:27.3067968+01:00|Found Signature [ServerTime] at offset [01D6FD70]|ec05fb03c2adee27067209f19e1327ea
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 5: 251|2020-12-28T19:57:27.3549255+01:00|Found Signature [Target] at offset [01D8A1C0]|49a7cba070b5be5c42265a6b664ff0ae
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 6: 251|2020-12-28T19:57:27.3584901+01:00|Found Signature [ChatLog] at offset [016AEEB8]|687c5917f205d546168bf660ae7311f8
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 7: 251|2020-12-28T19:57:27.3647289+01:00|Found Signature [Player] at offset [01DB5758]|dcd703cf91511baf53926d760a9e1eb7
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 8: 251|2020-12-28T19:57:27.3647289+01:00|Found Signature [PartyList] at offset [01DDAA20]|e539d8632d9cc51c9cad0dc471eedf60
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 9: 251|2020-12-28T19:57:27.3870092+01:00|Found Signature [ZoneID] at offset [01D8A00C]|d5425b467e0a4d2324001e88ed4bd3f0
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 10: 251|2020-12-28T19:57:27.3985951+01:00|Found Signature [MobArray] at offset [01DB2050]|c3651597a32096445a41caa1d9c90917
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 11: 251|2020-12-28T19:57:28.5074019+01:00|Found VTable for [ChatLog] at address [14A7E32EB90]|310797514b9038e761841d60d5bb51e2
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 12: 250|2020-12-28T19:57:28.5118584+01:00|Process ID: 51648, Client Mode: FFXIV_64, IsAdmin: True, Game Version: 2020.12.15.0000.0000|c932f5a9593f89960501e728d4a2f049
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 13: 251|2020-12-28T19:57:28.5172056+01:00|ProcessTCPInfo: New connection detected for Process [51648]: 192.168.2.46:34042=>185.199.108.153:443|817343de52aa1ece049c154cd0eb568b
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 14: 251|2020-12-28T19:57:28.5172056+01:00|ProcessTCPInfo: New connection detected for Process [51648]: 192.168.2.46:34047=>162.159.133.233:443|b60ffc8c2327e8b88856a0dad0dda02b
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 15: 251|2020-12-28T19:57:28.5172056+01:00|ProcessTCPInfo: New connection detected for Process [51648]: 192.168.2.46:34050=>173.195.146.142:443|f45279bab49fa971112d0ce6a3f10d96
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 16: 251|2020-12-28T19:57:28.5172056+01:00|ProcessTCPInfo: New connection detected for Process [51648]: 192.168.2.46:34051=>151.101.12.133:443|ff56d31bc5b7027ca99c6101b09188b9
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 17: 251|2020-12-28T19:57:28.5172056+01:00|ProcessTCPInfo: New connection detected for Process [51648]: 192.168.2.46:34160=>195.82.50.56:55024|8e3452f0c94a1977b75479a0cd69090e
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 18: 00|2020-12-28T19:57:28.0936021+01:00|0000|Hojoring|[EX] Beacon X=21,50 Y=21,50 Z=0,00 zone=Mist|
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 19: 251|2020-12-28T19:57:28.5172056+01:00|ProcessTCPInfo: New connection detected for Process [51648]: 192.168.2.46:34161=>195.82.50.56:55024|cabf9009be43f47f18891bb18552b88a
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 20: 251|2020-12-28T19:57:28.5207702+01:00|TCPNetworkMonitor: Starting RawSocket listener on [192.168.2.46]=> [185.199.108.153].|f75c6451015614a7c66f1d2c93961ac4
[28.12.2020 19:57:29] Info: LoadAddons: CactbotOverlay.dll: Initialized Cactbot.PluginLoader
[28.12.2020 19:57:29] Info: DEBUG: LogLineHandler: line 21: 01|2020-12-28T19:57:28.5810000+01:00|86|Middle La Noscea|79214df19ec8e6082580b01be742a789
[28.12.2020 19:57:30] Info: WS: Warn: The current logging level has been changed to Info. 
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 21: ChangeZone line, length 5
[28.12.2020 19:57:31] Info: DEBUG: DispatchAndCacheEvent: ChangeZone: Middle La Noscea
[28.12.2020 19:57:31] Info: DEBUG: DispatchEvent: ChangeZone, Middle La Noscea
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 22: 02|2020-12-28T19:57:28.5810000+01:00|107e1669|Kamo Akurosia|cabd20aea514e408cff8f4cf3179d6fc
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 23: 03|2020-12-28T19:57:28.5810000+01:00|107e1669|Kamo Akurosia|23|50|0|43|Shiva|0|0|112269|112269|10000|10000|0|0|223.933|-260.687|113.1|0.7195916||8fa13a811d0ce36f727f39f27ab237dd
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 24: 251|2020-12-28T19:57:28.5350304+01:00|TCPNetworkMonitor: Starting RawSocket listener on [192.168.2.46]=> [162.159.133.233].|0495a2689a797c2499f30c55ed3c38b3
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 25: 03|2020-12-28T19:57:28.5810000+01:00|107169c5|Riahin Kispoor|1d|12|0|43|Shiva|0|0|352|352|10000|10000|0|0|215.9382|-236.2919|127.5288|-2.989535||5d65d0e03f98ac4710d27f3c71ce94aa
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 26: 03|2020-12-28T19:57:28.5810000+01:00|400007c0|Trainingspuppe|0|1|0|0||541|901|44|44|0|10000|0|0|249.153|-208.7466|105.7127|-2.56893||9c4a1eccad89577ca00609c443fe528d
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 27: 03|2020-12-28T19:57:28.5810000+01:00|400007bf|Trainingspuppe|0|1|0|0||541|901|44|44|0|10000|0|0|253.5589|-214.0755|105.9843|-2.434513||2409f2ca3abaa0c950f1a6a568d4368c
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 28: 03|2020-12-28T19:57:28.5810000+01:00|400007c1|Trainingspuppe|0|1|0|0||541|901|44|44|0|10000|0|0|244.7265|-204.803|105.5582|-2.608622||d92ffc910ea37aa1db390c3886b3fa71
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 29: 03|2020-12-28T19:57:28.5810000+01:00|400007c2|Trainingspuppe|0|1|0|0||541|901|44|44|0|10000|0|0|220.353|-192.0823|102.2575|-3.141593||0051712370707aa0f49e7522f84a6a50
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 30: 03|2020-12-28T19:57:28.5810000+01:00|400007be|Trainingspuppe|0|1|0|0||541|901|44|44|0|10000|0|0|226.0351|-192.896|103.208|-2.948212||93db69d3dd72eece21a40b189dc44bb3
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 31: 03|2020-12-28T19:57:28.5810000+01:00|40048231|Winzige Mandragora|0|5|0|0||405|118|91|91|0|10000|0|0|145.983|-175.2804|83.44177|-0.5933242||74d6c410a7746acad26c2c04ad1933c5
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 32: 03|2020-12-28T19:57:28.5810000+01:00|4004a95e|Bienenwolke|0|5|0|0||395|57|91|91|0|10000|0|0|132.0668|-219.6841|84.64523|1.653896||441cdeb9a6b115632120b7abc4c0a46f
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 33: 03|2020-12-28T19:57:28.5810000+01:00|4004b1e8|Winzige Mandragora|0|5|0|0||405|118|91|91|0|10000|0|0|130.5191|-214.454|83.68782|2.584221||7de1e3f96cab057c15471bcf4f3a5bc9
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 34: 03|2020-12-28T19:57:28.5810000+01:00|4004a960|Winzige Mandragora|0|5|0|0||405|118|91|91|0|10000|0|0|133.7585|-216.9255|85.24932|2.279962||21d75cba80e9483cdee0f82f8555b381
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 35: 12|2020-12-28T19:57:28.5810000+01:00|35|186|358|5211|5232|377|340|186|3380|3384|5232|377|1567|380|606|0|380|40002E92F42814|966c864abfc1190d4e520b47f31d5fdb
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 36: 251|2020-12-28T19:57:28.5430510+01:00|TCPNetworkMonitor: Starting RawSocket listener on [192.168.2.46]=> [173.195.146.142].|ad13e2ceb099fbc79ca227f6efd16c4a
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 37: 251|2020-12-28T19:57:28.5611228+01:00|TCPNetworkMonitor: Starting RawSocket listener on [192.168.2.46]=> [151.101.12.133].|c61f2fe2bdb98db52aed48aac3ec4930
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 38: 251|2020-12-28T19:57:28.5688964+01:00|TCPNetworkMonitor: Starting RawSocket listener on [192.168.2.46]=> [195.82.50.56].|ff695a22680baef665f6c2d8c772c558
[28.12.2020 19:57:31] Info: DEBUG: LogLineHandler: line 39: 39|2020-12-28T19:57:29.3540000+01:00|107169C5|Riahin Kispoor|352|352|10000|10000|0|0|215.9382|-236.2919|127.5288|-2.989535||04ec75fd214ae3555b1c5ed6292e65f1
[28.12.2020 19:57:31] Info: timeline: BrowserConsole: local user file: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\user\\raidboss.js (Source: file:///E:/FINAL%20FANTASY%20XIV%20-%20A%20Realm%20Reborn/AdditionalExtern/ACT/Plugins/OverlayPlugin_ngld/cactbot/dist/raidboss.bundle.js, Line: 1)
[28.12.2020 19:57:31] Info: timeline: BrowserConsole: local user file: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\user\\raidboss.css (Source: file:///E:/FINAL%20FANTASY%20XIV%20-%20A%20Realm%20Reborn/AdditionalExtern/ACT/Plugins/OverlayPlugin_ngld/cactbot/dist/raidboss.bundle.js, Line: 1)
[28.12.2020 19:57:31] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 19:57:32] Info: DEBUG: StateCallback: ChangeZone, eventCache: Middle La Noscea
[28.12.2020 19:57:32] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 19:57:32] Info: DEBUG: StateCallback: ChangeZone, eventCache: Middle La Noscea
[28.12.2020 19:57:32] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 19:57:32] Info: DEBUG: StateCallback: ChangeZone, eventCache: Middle La Noscea
[28.12.2020 19:57:32] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 19:57:32] Info: DEBUG: StateCallback: ChangeZone, eventCache: Middle La Noscea
[28.12.2020 19:57:32] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 19:57:32] Info: DEBUG: StateCallback: ChangeZone, eventCache: Middle La Noscea
[28.12.2020 19:57:32] Info: timeline: BrowserConsole: Loading 00-misc/general.js (Source: file:///E:/FINAL%20FANTASY%20XIV%20-%20A%20Realm%20Reborn/AdditionalExtern/ACT/Plugins/OverlayPlugin_ngld/cactbot/dist/raidboss.bundle.js, Line: 1)
[28.12.2020 19:57:32] Info: timeline: BrowserConsole: Loading 00-misc/test.js (Source: file:///E:/FINAL%20FANTASY%20XIV%20-%20A%20Realm%20Reborn/AdditionalExtern/ACT/Plugins/OverlayPlugin_ngld/cactbot/dist/raidboss.bundle.js, Line: 1)
[28.12.2020 19:57:32] Info: Found enmity memory for 5.4.
```

**Comment by Akurosia** [comment]
Funny enough, right now it is working (tried it again 3 times and it detected the instance) on 2 different chars.

**Comment by Akurosia** [comment]
and now again it is not working 
```
[28.12.2020 20:24:57] Info: AssemblyResolver: Loaded: Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed
[28.12.2020 20:24:57] Info: AssemblyResolver: Loaded: websocket-sharp, Version=1.0.2.32519, Culture=neutral, PublicKeyToken=5660b08a1845a91e
[28.12.2020 20:24:57] Info: AssemblyResolver: Loaded: HtmlRenderer, Version=0.16.2.0, Culture=neutral, PublicKeyToken=null
[28.12.2020 20:24:57] Info: AssemblyResolver: Loaded: CefSharp, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[28.12.2020 20:24:57] Info: InitPlugin: PluginDirectory = E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld
[28.12.2020 20:24:57] Info: AssemblyResolver: Loaded: CefSharp.OffScreen, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[28.12.2020 20:24:57] Info: AssemblyResolver: Loaded: CefSharp.Core, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[28.12.2020 20:24:58] Info: AssemblyResolver: Loaded: OverlayPlugin.Core.resources, Version=0.16.2.0, Culture=de-DE, PublicKeyToken=null
[28.12.2020 20:24:58] Info: InitPlugin: Initialised.
[28.12.2020 20:25:05] Info: DEBUG: RegisterCachedEventType: ChangeZone
[28.12.2020 20:25:05] Info: cactbot: 0.21.10.0 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\CactbotOverlay.dll (dir: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot)
[28.12.2020 20:25:05] Info: OverlayPlugin: 0.16.2.0 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\OverlayPlugin.dll
[28.12.2020 20:25:05] Info: FFXIV Plugin: 2.0.7.4 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\FFXIV_ACT_Plugin.dll
[28.12.2020 20:25:05] Info: ACT: 3.5.0.273 E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Advanced Combat Tracker.exe
[28.12.2020 20:25:05] Info: Parsing Plugin Language: de
[28.12.2020 20:25:05] Info: System Locale: de-DE
[28.12.2020 20:25:05] Info: cactbot user directory: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\user\
[28.12.2020 20:25:05] Info: Version: intl
[28.12.2020 20:25:05] Info: LoadAddons: CactbotOverlay.dll: Initialized Cactbot.PluginLoader
[28.12.2020 20:25:05] Info: DEBUG: LogLineHandler: line 1: 39|2020-12-28T20:25:06.2290000+01:00|10765874|Haqpot Arulaq|49556|49556|10000|10000|0|0|133.2252|-204.5441|84.29442|-1.28075||7932c5450a2da20b57e0e65ec1575c73
[28.12.2020 20:25:06] Info: DEBUG: LogLineHandler: line 2: 39|2020-12-28T20:25:06.9850000+01:00|10778807|Phileas Ashan|112335|112335|10000|10000|0|0|222.6668|-264.7649|112.9391|2.453783||71c0305f43dd3ba510823598f1c10aea
[28.12.2020 20:25:06] Info: WS: Warn: The current logging level has been changed to Info. 
[28.12.2020 20:25:08] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 20:25:08] Info: DEBUG: StateCallback: ChangeZone, eventCache: null
[28.12.2020 20:25:08] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 20:25:08] Info: DEBUG: StateCallback: ChangeZone, eventCache: null
[28.12.2020 20:25:08] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 20:25:08] Info: DEBUG: StateCallback: ChangeZone, eventCache: null
[28.12.2020 20:25:08] Info: timeline: BrowserConsole: local user file: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\user\\raidboss.js (Source: file:///E:/FINAL%20FANTASY%20XIV%20-%20A%20Realm%20Reborn/AdditionalExtern/ACT/Plugins/OverlayPlugin_ngld/cactbot/dist/raidboss.bundle.js, Line: 1)
[28.12.2020 20:25:08] Info: timeline: BrowserConsole: local user file: E:\FINAL FANTASY XIV - A Realm Reborn\AdditionalExtern\ACT\Plugins\OverlayPlugin_ngld\cactbot\user\\raidboss.css (Source: file:///E:/FINAL%20FANTASY%20XIV%20-%20A%20Realm%20Reborn/AdditionalExtern/ACT/Plugins/OverlayPlugin_ngld/cactbot/dist/raidboss.bundle.js, Line: 1)
[28.12.2020 20:25:08] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 20:25:08] Info: DEBUG: StateCallback: ChangeZone, eventCache: null
[28.12.2020 20:25:08] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 20:25:08] Info: DEBUG: StateCallback: ChangeZone, eventCache: null
[28.12.2020 20:25:08] Info: DEBUG: Subscribe: ChangeZone, getting cached state
[28.12.2020 20:25:08] Info: DEBUG: LogLineHandler: line 3: 39|2020-12-28T20:25:09.2640000+01:00|10765874|Haqpot Arulaq|49556|49556|10000|10000|0|0|133.2252|-204.5441|84.29442|-1.28075||34922028e4aa30468244ac74ad4428f1
[28.12.2020 20:25:08] Info: DEBUG: StateCallback: ChangeZone, eventCache: null
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 4: 04|2020-12-28T20:25:09.3580000+01:00|4004bb07|Bienenwolke|0|5|0|0||395|57|91|91|0|10000|0|0|130.2656|-216.6983|83.59894|-1.42207||a973d3090c17482a39c770a3cee4b81b
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 5: 38|2020-12-28T20:25:09.8030000+01:00|10680E8D|Ergdar'li Mulhazi|00080803|160|182|10000|10000|0|0|196.7207|-213.0842|108.589|-1.703172|0|0|0||aeb94137550025c08c806e288148a941
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 6: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 99%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 7: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 98%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 8: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 97%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 9: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 96%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 10: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 95%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 11: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 94%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 12: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 93%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 13: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 92%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 14: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 91%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 15: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 90%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 16: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 89%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 17: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 88%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 18: 13|2020-12-28T20:25:09.4520000+01:00|Ergdar'li Mulhazi HP at 87%.
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 19: 39|2020-12-28T20:25:09.9800000+01:00|10778807|Phileas Ashan|112335|112335|10000|10000|0|0|222.6668|-264.7649|112.9391|2.453783||7ace8eb96e610ab6f9d0c82ab60bd48f
[28.12.2020 20:25:09] Info: DEBUG: LogLineHandler: line 20: 00|2020-12-28T20:25:09.0000000+01:00|2229||   Ergdar'li Mulhazi erleidet 22 Punkte Schaden.|d8fda3f3a7e2935dbda29330855ad7c1
[28.12.2020 20:25:11] Info: DEBUG: LogLineHandler: line 21: 39|2020-12-28T20:25:11.5380000+01:00|10680E8D|Ergdar'li Mulhazi|178|182|10000|10000|0|0|187.6264|-210.5678|97.56281|-1.394016||a5fe5bbe81f58d1f885923b20ee1ed04
[28.12.2020 20:25:11] Info: DEBUG: LogLineHandler: line 22: 13|2020-12-28T20:25:11.1960000+01:00|Ergdar'li Mulhazi HP at 97%.
[28.12.2020 20:25:11] Info: DEBUG: LogLineHandler: line 23: 39|2020-12-28T20:25:12.2510000+01:00|10765874|Haqpot Arulaq|49556|49556|10000|10000|0|0|133.2252|-204.5441|84.29442|-1.28075||0991a4493601ab4c1ec5d548ddfeaddf
[28.12.2020 20:25:12] Info: DEBUG: LogLineHandler: line 24: 39|2020-12-28T20:25:13.0090000+01:00|10778807|Phileas Ashan|112335|112335|10000|10000|0|0|222.6668|-264.7649|112.9391|2.453783||ab14c5617ae6dbef9a7a50dd588d6c36
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 25: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Akurosia Kamo X=26,25 Y=17,04 Z=1,13 hp=113108|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 26: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Trainingspuppe X=26,48 Y=17,33 Z=1,06 hp=44|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 27: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Riahin Kispoor X=25,82 Y=16,77 Z=1,28 hp=352|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 28: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Trainingspuppe X=26,39 Y=17,40 Z=1,06 hp=44|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 29: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Trainingspuppe X=26,57 Y=17,22 Z=1,06 hp=44|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 30: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Mia Mikada X=25,63 Y=17,04 Z=1,13 hp=200|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 31: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Trainingspuppe X=25,91 Y=17,66 Z=1,02 hp=44|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 32: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Trainingspuppe X=26,02 Y=17,64 Z=1,03 hp=44|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 33: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Phileas Ashan X=25,95 Y=16,20 Z=1,13 hp=112335|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 34: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Winzige Mandragora X=24,34 Y=17,86 Z=0,84 hp=91|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 35: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Winzige Mandragora X=24,09 Y=17,23 Z=0,83 hp=91|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 36: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Haqpot Arulaq X=24,16 Y=17,41 Z=0,84 hp=49556|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 37: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Ergdar'li Mulhazi X=25,04 Y=17,30 Z=0,96 hp=178|
[28.12.2020 20:25:13] Info: DEBUG: LogLineHandler: line 38: 00|2020-12-28T20:25:13.3721543+01:00|0000|Hojoring|[EX] Added new combatant. name=Winzige Mandragora X=24,19 Y=17,31 Z=0,85 hp=91|
[28.12.2020 20:25:14] Info: DEBUG: LogLineHandler: line 39: 39|2020-12-28T20:25:14.5240000+01:00|10680E8D|Ergdar'li Mulhazi|182|182|10000|10000|0|0|172.3139|-209.5533|95.24895|-1.386065||1743fd2d3799e612cb50b6251e9d9af8
```

**Comment by quisquous** [comment]
So, when I run this build, I get the following:
```
[12/29/2020 10:22:50 AM] Info: AssemblyResolver: Loaded: Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed
[12/29/2020 10:22:50 AM] Info: AssemblyResolver: Loaded: websocket-sharp, Version=1.0.2.32519, Culture=neutral, PublicKeyToken=5660b08a1845a91e
[12/29/2020 10:22:50 AM] Info: AssemblyResolver: Loaded: HtmlRenderer, Version=0.16.2.0, Culture=neutral, PublicKeyToken=null
[12/29/2020 10:22:50 AM] Info: AssemblyResolver: Loaded: CefSharp, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[12/29/2020 10:22:50 AM] Info: InitPlugin: PluginDirectory = C:\Users\tinipoutini\OverlayPlugin.ngld\out\Release
[12/29/2020 10:22:50 AM] Info: AssemblyResolver: Loaded: CefSharp.OffScreen, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[12/29/2020 10:22:50 AM] Info: AssemblyResolver: Loaded: CefSharp.Core, Version=75.1.141.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138
[12/29/2020 10:22:51 AM] Info: InitPlugin: Initialised.
[12/29/2020 10:22:52 AM] Info: DEBUG: RegisterCachedEventType: ChangeZone
[12/29/2020 10:22:52 AM] Info: DEBUG: LogLineHandler: line 1: 249|2020-12-29T10:22:50.0352916-08:00|Language ID: 1, Disable Damage Shield: False, Disable Combine Pets: False, Parse Filter: Alliance, DoTCrits: False, RealDoTs: False, Parse Potency: False|65b1a79e590b8b9d6e17bcd91413fb17
[12/29/2020 10:22:53 AM] Info: DEBUG: LogLineHandler: line 2: 249|2020-12-29T10:22:50.0622932-08:00|Selected Process ID: 0, Disable Network Data: False, Enable HP Scan: True, Dump All Network Data: False, Disable Combat Log: False, Selected IP: , WinPcap: False, Socket Filter: True|6a345127927fd771402fa8075ffa3af6
[12/29/2020 10:22:53 AM] Info: DEBUG: LogLineHandler: line 3: 251|2020-12-29T10:22:52.9744597-08:00|Found Signature [ServerTime] at offset [01D6FD70]|fe8ef720a3d64d284794add0e08e342a
[12/29/2020 10:22:53 AM] Info: DEBUG: LogLineHandler: line 4: 251|2020-12-29T10:22:53.0524642-08:00|Found Signature [Target] at offset [01D8A1C0]|77b27441263051cdbc06bb770679a056
[12/29/2020 10:22:53 AM] Info: DEBUG: LogLineHandler: line 5: 251|2020-12-29T10:22:53.0584645-08:00|Found Signature [ChatLog] at offset [016AEEB8]|7e9246a7f5e26d914de5445cd89dec51
[12/29/2020 10:22:53 AM] Info: DEBUG: LogLineHandler: line 6: 251|2020-12-29T10:22:53.0734654-08:00|Found Signature [Player] at offset [01DB5758]|b359e37cdc7a832fb0ad48e478deda24
[12/29/2020 10:22:53 AM] Info: DEBUG: LogLineHandler: line 7: 251|2020-12-29T10:22:53.0734654-08:00|Found Signature [PartyList] at offset [01DDAA20]|70a287391154d58617b905f55b58ab69
[12/29/2020 10:22:53 AM] Info: cactbot: 0.21.10.0 C:\Users\tinipoutini\cactbot\bin\x64\Debug\CactbotOverlay.dll (dir: C:\Users\tinipoutini\cactbot\)
[12/29/2020 10:22:53 AM] Info: OverlayPlugin: 0.16.2.0 C:\Users\tinipoutini\OverlayPlugin.ngld\out\Release\OverlayPlugin.dll
[12/29/2020 10:22:53 AM] Info: FFXIV Plugin: 2.0.7.4 C:\Users\tinipoutini\AppData\Roaming\Advanced Combat Tracker\Plugins\FFXIV_ACT_Plugin.dll
[12/29/2020 10:22:53 AM] Info: ACT: 3.5.0.273 C:\Program Files (x86)\Advanced Combat Tracker\Advanced Combat Tracker.exe
[12/29/2020 10:22:53 AM] Info: Parsing Plugin Language: en
[12/29/2020 10:22:53 AM] Info: System Locale: en-US
[12/29/2020 10:22:53 AM] Info: cactbot user directory: C:\Users\tinipoutini\cactbot\user\
[12/29/2020 10:22:53 AM] Info: Version: intl
[12/29/2020 10:22:53 AM] Info: Ignoring auto-update due to cactbot directory being a .git repo.
[12/29/2020 10:22:53 AM] Info: DEBUG: LogLineHandler: line 8: 251|2020-12-29T10:22:53.1094675-08:00|Found Signature [ZoneID] at offset [01D8A00C]|b10af6c8105a4b2975de43a08c68ca56
[12/29/2020 10:22:53 AM] Info: LoadAddons: CactbotOverlay.dll: Initialized Cactbot.PluginLoader
[12/29/2020 10:22:53 AM] Info: DEBUG: LogLineHandler: line 9: 251|2020-12-29T10:22:53.1364690-08:00|Found Signature [MobArray] at offset [01DB2050]|de0ae9d04e18fd10033f774161876882
[12/29/2020 10:22:54 AM] Info: WS: Warn: The current logging level has been changed to Info. 
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 10: 251|2020-12-29T10:22:54.1805287-08:00|Found VTable for [ChatLog] at address [34B017B0]|afb3422711fbffabf98304f116e71dc1
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 11: 250|2020-12-29T10:22:54.2145307-08:00|Process ID: 5640, Client Mode: FFXIV_64, IsAdmin: True, Game Version: 2020.12.15.0000.0000|885bb873d832103129cb89076d07b5f5
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 12: 01|2020-12-29T10:22:55.4540000-08:00|153|Mist|70f6ef9acd125caa4b69b704394b64f8
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 12: ChangeZone line, length 5
[12/29/2020 10:22:54 AM] Info: DEBUG: DispatchAndCacheEvent: ChangeZone: Mist
[12/29/2020 10:22:54 AM] Info: DEBUG: DispatchEvent: ChangeZone, Mist
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 13: 251|2020-12-29T10:22:54.2345318-08:00|ProcessTCPInfo: New connection detected for Process [5640]: 192.168.1.70:49387=>204.2.229.85:55007|e1e745842ff23ff289a1e023e1b432f0
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 14: 251|2020-12-29T10:22:54.2345318-08:00|ProcessTCPInfo: New connection detected for Process [5640]: 192.168.1.70:49388=>204.2.229.85:55007|1f02a390f981752990784d27c47f4e70
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 15: 251|2020-12-29T10:22:54.2405321-08:00|TCPNetworkMonitor: Starting RawSocket listener on [192.168.1.70]=> [204.2.229.85].|4c85a3ca6af6f3b8f8da85157707ffee
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 16: 02|2020-12-29T10:22:55.4540000-08:00|12345678|Tini Poutini|8e2c17bb846a833aa4cbf9d4601f9917
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 17: 03|2020-12-29T10:22:55.4540000-08:00|12345678|Tini Poutini|a|50|0|28|Jenova|0|0|4731|4731|0|10000|0|0|-659.8143|-843.3312|29.80632|-1.36204||0b44d106725757552f5bebdb2831631b
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 18: 03|2020-12-29T10:22:55.4540000-08:00|40001212|Striking Dummy|0|50|0|0||541|10183|7400000|7400000|0|10000|0|0|-632.7191|-845.134|35.99999|1.570772||324621ffa0845dfc15311ae125988479
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 19: 03|2020-12-29T10:22:55.4540000-08:00|400011e1|Striking Dummy|0|32|0|0||541|901|2778|2778|0|10000|0|0|-705.9282|-809.8726|18|-2.356566||799c10107b9cc819876624aa2dc7ae7d
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 20: 03|2020-12-29T10:22:55.4540000-08:00|400011e2|Striking Dummy|0|50|0|0||541|10183|7400000|7400000|0|10000|0|0|-678.3286|-786.0344|18.00002|1.570389||9109c4f43648b62e0804aaf348bf0618
[12/29/2020 10:22:54 AM] Info: DEBUG: LogLineHandler: line 21: 03|2020-12-29T10:22:55.4540000-08:00|40001220|Striking Dummy|0|50|0|0||541|10183|7400000|7400000|0|10000|0|0|-581.8538|-789.5431|40.00002|-0.0004315376||69bdbdf075026be8942ffd9065ff3bb4
```

In particular, I get all of the `251` debug lines followed by the initial `01` (zone) then `02` (current player) that always starts the log file.  It looks like there's maybe a race condition here between when ffxiv starts emitting log lines and when overlayplugin starts listening to them.

Do you have hojoring between the ffxiv plugin and overlayplugin, or some other weird plugin ordering? If you disable hojoring, does it work properly?

**Comment by Akurosia** [comment]
my active plugins (in order) are: FFXIV Plugin, OverlayPlugin, Cactbot, Hojuring SpecialSpellTimer, Hojuring Yukkuri, Hojuring Ultrascouter, Triggernometry

Also since the last few weeks nothing has changed in there in regards of order of these plugins and before 5.4 i had never any issues like this before.

**Comment by Akurosia** [comment]
Ok i disabled all of hojuring and tried it a couple of times and sofar its working like quis described it (all 251 lines followed by 01 and 02)

i will have a detailed look at it during the next days and see what happens

**Comment by Akurosia** [comment]
Ok after some days of testing, it really has to do with Hojouring. Its a like 80:20 game of notworking vs working when starting ACT.

**Comment by ngld** [comment]
Does this happen if you launch ACT first and FFXIV afterwards or does it only happen if you launch FFXIV first?
Most likely this is a race condition. The FFXIV plugin starts sending log lines as soon as it's loaded. However, OverlayPlugin only starts receiving log lines once ACT is done loading plugins.
My theory is that the startup takes longer with Hojoring, long enough in fact that the initial zone log line is processed before OverlayPlugin is ready.

This is expected behavior. If you want the full and accurate game state in ACT, you have to start ACT first or change your zone at least once.

**Comment by Akurosia** [comment]
i always have FFXIV up and running when i did my tests.(by just closing ACT and reopening after a while to make sure its fully closed in task manager)

**Comment by quisquous** [comment]
Can OverlayPlugin subscribe to logs earlier?

I think the unexpected part for me is that `ChangeZoneEvent` is a cached event.  So, my expectation is that OverlayPlugin should always send this event as soon as I subscribe, if the game is on and I am therefore in some zone.  Similarly, the `ChangePrimaryPlayerEvent`.

**Comment by Akurosia** [comment]
just as a little side note: i also have trouvle with enmity memory when hojouring is loaded, when ACT starts. My current workaround is: disabling hojouring, restart act, wait till `Info: Found enmity memory for 5.4.` is displayed in the logs, enable hojouring again.

**Comment by ngld** [comment]
OverlayPlugin could subscribe to the logs earlier. However, since the event source subscribes to the log event, that'd require starting up the built in event sources earlier which would also mean that the built in event sources would have to start much earlier than i.e. Cactbot's event source which I'd rather avoid since it adds complexity and potential for errors.

Cached events only work if OverlayPlugin has the information in the first place. The only thing that's always available is memory data since that's read constantly. Log and network based data are only available once OverlayPlugin has seen the relevant packets / log lines.

I could add a disclaimer that some data might be missing until you change your zone if you start ACT after FFXIV though I'm not sure where I'd put that disclaimer (in the dev docs? The UI?). I'm not really willing to change this either since it's a natural consequence of the parser design and will always remain an issue as long as we rely on raw network data anywhere. I also doubt the effort required to fix this is worth saving the player a teleport.

@Akurosia Can you describe your issue in more detail? Do you get errors or any other messages related to memory? How long does it take for the `Found enmity memory` message to show up?

**Comment by quisquous** [comment]
> OverlayPlugin could subscribe to the logs earlier.

What about subscribing to logs earlier with some other handler and stashing them somewhere until everything else is set up, and then processing everything then?

I agree that log data is only available once OverlayPlugin has seen the log line, but these aren't log lines like combat data that existed five minutes ago, they are log lines that get sent every time when the parser plugin starts writing to a log file.  It feels like these shouldn't be missed and OverlayPlugin is missing them due to a race.

I guess arguably you could enable OverlayPlugin after ACT startup.  Maybe we could get some way to ask the parser plugin to resend all the header logs that start every log file.

> I also doubt the effort required to fix this is worth saving the player a teleport.

Sorry to push back on this, but many people zone into a raid and then start ACT.  Additionally, sometimes your computer crashes and you need to restart ACT while you're in a zone.  It's not clear when you're in this state, and you're not always in a place where you can teleport.

**Comment by ngld** [comment]
If we buffer the log lines, the `ChangePrimaryPlayer` and `ChangeZone` events would work as expected but it won't fix the `OnlineStatusChanged` and `PartyChanged` events. To fix the `PartyChanged` event, I'd have to also register the `PartyChangeDelegate` early. At that point it's probably easier to start the event source earlier. Finally, there's nothing I can do to fix the `OnlineStatusChanged` event but I doubt anyone cares about that (since it's mostly used for cutscene detection).

If I register the `PartyChangeDelegate` early, loading OverlayPlugin after the FFXIV plugin becomes a hard requirement but most people seem to do that anyway. I can probably display a warning if I detect that OverlayPlugin is loaded before the FFXIV plugin to work around that issue.

I'll probably have to split the load phases, then. The FFXIV integration has to be enabled early so we can register the `PartyChangeDelegate` but the Trigg integration has to be enabled late because I can't rely on people loading Trigg before OverlayPlugin. Nothing uses the Trigg integration directly so that should be fine.

To register the event handlers, it's enough to create the event source instances early, I don't have to start them so it shouldn't mess with other stuff like the config UI.
This wouldn't cause Cactbot's own event source to start any earlier but I don't think it's affected because it doesn't listen to any ACT or FFXIV events in the first place.

**Comment by Akurosia** [comment]
> @Akurosia Can you describe your issue in more detail? Do you get errors or any other messages related to memory? How long does it take for the `Found enmity memory` message to show up?

I do not get any kind of errors in my log and i also dont get any `Found enmity memory` in like the 20-30min i tested it
